tslint.json

```diff
@@ -1,183 +1,124 @@
 {
-  "extends": [
-    "tslint:recommended",
-    "rxjs-tslint-rules"
-  ],
+  "extends": ["tslint:latest", "tslint-config-prettier", "rxjs-tslint-rules"],
+  "rulesDirectory": ["codelyzer"],
   "rules": {
-    "align": {
-      "options": [
-        "parameters",
-        "statements"
-      ]
-    },
-    "array-type": false,
-    "arrow-return-shorthand": true,
-    "curly": true,
-    "deprecation": {
-      "severity": "warning"
-    },
-    "component-class-suffix": [
-      true,
-      "Page",
-      "Component"
-    ],
-    "contextual-lifecycle": true,
-    "directive-class-suffix": true,
-    "directive-selector": [
-      true,
-      "attribute",
-      "app",
-      "camelCase"
-    ],
-    "component-selector": [
-      true,
-      "element",
-      "app",
-      "kebab-case"
-    ],
-    "eofline": true,
-    "import-blacklist": [
-      true,
-      "rxjs/Rx"
-    ],
-    "import-spacing": true,
-    "indent": {
-      "options": [
-        "spaces"
-      ]
-    },
-    "max-classes-per-file": false,
-    "max-line-length": [
-      true,
-      140
-    ],
     "member-ordering": [
       true,
       {
         "order": [
-          "static-field",
           "instance-field",
-          "static-method",
-          "instance-method"
+          "constructor",
+          "instance-method",
+          "static-field",
+          "static-method"
         ]
       }
     ],
-    "no-console": [
-      true,
-      "debug",
-      "info",
-      "time",
-      "timeEnd",
-      "trace"
-    ],
-    "no-empty": false,
-    "no-inferrable-types": [
-      true,
-      "ignore-params"
-    ],
+    "no-for-in": true,
+    "no-inferrable-types": true,
+    "no-magic-numbers": [true, -1, 0, 1, 2, 3],
     "no-non-null-assertion": true,
-    "no-redundant-jsdoc": true,
+    "no-parameter-reassignment": true,
+    "no-unnecessary-type-assertion": true,
+    "promise-function-async": true,
+    "await-promise": true,
+    "ban-comma-operator": true,
+    "import-blacklist": [true, "rxjs/Rx"],
+    "no-async-without-await": true,
+    "no-console": [true, "log", "info", "debug"],
+    "no-duplicate-switch-case": true,
+    "no-for-in-array": true,
+    "no-inferred-empty-object-type": true,
+    "no-invalid-this": true,
+    "no-null-keyword": true,
+    "no-promise-as-boolean": true,
+    "no-submodule-imports": false,
     "no-switch-case-fall-through": true,
-    "no-var-requires": false,
```

Comment:
Why remove `no-var-requires` rule?

Reply:
Many third-party libraries still do not support ES6 module. Thus, we still need to use `require()` to import the library. Though we remove the linter warning, Angular compiler will still warn that using the syntax of CommonJS to import module will cause the ES6 import optimization fail. But if you think this rule is necessary for linting, I can add it back.